src/test/kotlin/io/github/facilityapi/intellij/inspection/InvalidNullableTypeInspectionTest.kt

```diff
@@ -0,0 +1,32 @@
+package io.github.facilityapi.intellij.inspection
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.intellij.codeInsight.daemon.impl.HighlightInfo
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import io.github.facilityapi.intellij.FsdLanguage
+
+class InvalidNullableTypeInspectionTest : BasePlatformTestCase() {
+    fun `test recursive nullability detected`() {
+        val code = """
+        service MessageService {
+            data Message
+            {
+                id: string;
+                text: nullable<nullable<string>>;
+            }
+        }
+        """
+
+        checkInspection(code, listOf("Nested nullable type: nullable<string>"))
```

Comment:
This error message is different than the C# parser, but I think it's clearer.

https://github.com/FacilityApi/Facility/blob/7e4ddef90bceb36b6e917e86e1f90e345458975a/tests/Facility.Definition.UnitTests/FieldTests.cs#L304